pkg/modules/relay/relayer.go

```diff
@@ -93,16 +94,34 @@ func (r *Relayer) SendUserOperation() modules.BatchHandlerFunc {
 		}
 		ctx.Data["relayer_est_revert_reasons"] = estRev
 
+		if len(ctx.Batch) == 0 {
+			return nil
+		}
+
+		// Accumulate the total gas limit of all user operations.
+		totalGasLimit := big.NewInt(0)
+		for _, op := range ctx.Batch {
+			totalGasLimit.Add(totalGasLimit, op.GetMaxGasAvailable())
+		}
+
+		// Estimated gas limit should be no less than the total gas limit, otherwise this transaction
+		// may be failed due to out of gas.
+		if opts.GasLimit <= totalGasLimit.Uint64() {
+			opts.GasLimit = totalGasLimit.Uint64()
```

Comment:
Why not set `opts.GasLimit` correctly instead of modify here?

Reply:
Do you suggest to set `opts.GasLimit` to the total gas limit of all user operations directly without estimation first?

Reply:
I think that would work if the estimated gas limit never exceeds the total gas limit of all user opeartions, otherwises the bundle transaction might fail due to lack of gas.